test/project_types/script/script_project_test.rb

```diff
@@ -7,6 +7,29 @@ class ScriptProjectTest < MiniTest::Test
     def setup
       @context = TestHelpers::FakeContext.new
       @script_name = 'name'
+      @extension_point_type = 'ep_type'
+    end
+
+    def test_initialize
+      ScriptProject
+        .any_instance
+        .expects(:lookup_config)
+        .with('extension_point_type')
+        .returns(@extension_point_type)
+
+      ScriptProject
+        .any_instance
+        .expects(:lookup_config)
+        .with('script_name')
+        .returns(@script_name)
+
+      ScriptProject.new(directory: 'testdir')
+
+      assert_equal({
+        "script_name" => @script_name,
+        "extension_point_type" => @extension_point_type,
+        "language" => 'ts',
+      }, ShopifyCli::Core::Monorail.metadata)
```

Comment:
I think it will be better to use `ShopifyCli::Core::Monorail.expects(:metadata).with(...)` instead of actually calling the method in `Monorail` class.  In this case, it happens to be harmless, but what if calling that method actually has some side effect (such as writing some value into a file)?  We should isolate the actual dependencies.

Reply:
Since `Monorail.metadata` is an attribute and we're testing if its value has changed, I'm wondering if it makes more sense to keep it like this

Reply:
That sounds reasonable!